docs/polyglot/python.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: developer-doc
+title: Polyglot Python
+category: polyglot
+tags: [polyglot, python]
+order: 4
+---
+
+# Polyglot Python
+
+This document provides practical example showing polyglot interoperability with
+Python in the runtime. Please familiarise yourself with the general operation of
+[polyglot bindings](./polyglot-bindings.md).
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Polyglot Library System](#polyglot-library-system)
+- [Using Python Libraries](#using-python-libraries)
+
+<!-- /MarkdownTOC -->
+
+## Polyglot Library System
+
+There is a support for using any Python library from Enso. Steps to include
+`numpy` in a new Enso project follows:
+
+```bash
+$ enso-engine*/bin/enso new numenso
+$ find numenso/
+numenso/
+numenso/src
+numenso/src/Main.enso
+numenso/package.yaml
+$ mkdir numenso/polyglot
+$ graalvm/bin/graalpy -m venv numenso/polyglot/python
+$ source ./numenso/polyglot/python/bin/activate
+(python) ~$ pip install numpy
+Successfully installed numpy-1.23.5
+```
+
+The above steps instruct Enso to create a new project in `numenso` directory.
+Then they create Python virtual environment in `numenso/polyglot/python/` dir -
+e.g. in the
+[standard location for polyglot](../distribution/packaging.md#the-polyglot-directory)
+components of an Enso project. As a last step we activate the virtual
+environment and use `pip` manager to install `numpy` library.
+
+## Using Python Libraries
+
+As soon as a library is installed into the
+[polyglot directory](#polyglot-library-system) it can be used via the
+[embedded syntax](polyglot-bindings.md#embedded-syntax):
+
+```ruby
+foreign python random_array s = """
+    import numpy
+    return numpy.random.normal(size=s)
+
+main = random_array 10
+```
+
+Let's modify the `numenso/src/Main.enso` to use `numpy.random.normal` as shown
+above. Then we can simply execute the project and obtain a `numpy` array as a
+result:
+
+```bash
+$ enso-engine*/bin/enso --run numenso
```

Comment:
```
./built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/bin/enso --run numenso
Execution finished with an error: ModuleNotFoundError: No module named 'numpy'
        at <python> <python> polyglot_enso_python_eval(Unknown)
        at <epb> <epb> null(Unknown)
        at <enso> Main.random_array(src/Main.enso:6-8)
        at <enso> Main.main(src/Main.enso:11:5-19)
```
hmm, doesn't seem to work?

Reply:
What does debugger says at https://github.com/enso-org/enso/pull/7678/files#diff-0024a96ec0db8aac19bdc515cf0c8c108b0ac1f2beee916ad7da5e6737993805R98 ? Does it find the file? Maybe I messed something up on my local disk, but I swear I saw it working!

Reply:
OK. The problem was somewhere between the chair and the monitor.